zio-http/src/main/scala/zio/http/endpoint/openapi/JsonSchema.scala

```diff
@@ -0,0 +1,711 @@
+      case TextCodec.StringCodec      => JsonSchema.String
+      case TextCodec.IntCodec         => JsonSchema.Integer(JsonSchema.IntegerFormat.Int32)
+      case TextCodec.LongCodec        => JsonSchema.Integer(JsonSchema.IntegerFormat.Int64)
+      case TextCodec.BooleanCodec     => JsonSchema.Boolean
+      case TextCodec.UUIDCodec        => JsonSchema.String
+    }
+
+  def fromSegmentCodec(codec: SegmentCodec[_]): JsonSchema =
+    codec match {
+      case SegmentCodec.BoolSeg(_)                    => JsonSchema.Boolean
+      case SegmentCodec.IntSeg(_)                     => JsonSchema.Integer(JsonSchema.IntegerFormat.Int32)
+      case SegmentCodec.LongSeg(_)                    => JsonSchema.Integer(JsonSchema.IntegerFormat.Int64)
+      case SegmentCodec.Text(_)                       => JsonSchema.String
+      case SegmentCodec.UUID(_)                       => JsonSchema.String
+      case SegmentCodec.Annotated(codec, annotations) =>
+        fromSegmentCodec(codec).description(segmentDoc(annotations)).examples(segmentExamples(codec, annotations))
+      case SegmentCodec.Literal(_) => throw new IllegalArgumentException("Literal segment is not supported.")
+      case SegmentCodec.Empty      => throw new IllegalArgumentException("Empty segment is not supported.")
+      case SegmentCodec.Trailing   => throw new IllegalArgumentException("Trailing segment is not supported.")
+    }
+
+  private def segmentDoc(annotations: Chunk[SegmentCodec.MetaData[_]]) =
+    annotations.collect { case SegmentCodec.MetaData.Documented(doc) => doc }.reduceOption(_ + _).map(_.toCommonMark)
+
+  private def segmentExamples(codec: SegmentCodec[_], annotations: Chunk[SegmentCodec.MetaData[_]]) =
+    Chunk.fromIterable(
+      annotations.collect { case SegmentCodec.MetaData.Examples(example) => example.values }.flatten.map { value =>
+        codec match {
+          case SegmentCodec.Empty           => throw new IllegalArgumentException("Empty segment is not supported.")
+          case SegmentCodec.Literal(_)      => throw new IllegalArgumentException("Literal segment is not supported.")
+          case SegmentCodec.BoolSeg(_)      => Json.Bool(value.asInstanceOf[Boolean])
+          case SegmentCodec.IntSeg(_)       => Json.Num(value.asInstanceOf[Int])
+          case SegmentCodec.LongSeg(_)      => Json.Num(value.asInstanceOf[Long])
+          case SegmentCodec.Text(_)         => Json.Str(value.asInstanceOf[String])
+          case SegmentCodec.UUID(_)         => Json.Str(value.asInstanceOf[java.util.UUID].toString)
+          case SegmentCodec.Trailing        =>
+            throw new IllegalArgumentException("Trailing segment is not supported.")
+          case SegmentCodec.Annotated(_, _) =>
+            throw new IllegalStateException("Annotated SegmentCodec should never be nested.")
+        }
+      },
+    )
+
+  def fromZSchema(schema: Schema[_], refType: SchemaStyle = SchemaStyle.Inline): JsonSchema =
```

Comment:
Would it be beneficial to support new `zio.schema.annotation.description` here like https://github.com/zio/zio-http/pull/2499/ ?

Reply:
I'd leave this PR as is and add it later.